src/Datadog.Trace/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Linq;
+
+namespace Datadog.Trace.Plugins
+{
+    internal static class PluginManager
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(PluginManager));
+
+        internal static IReadOnlyCollection<IOTelPlugin> TryLoadPlugins(JsonConfigurationSource pluginsConfig)
+        {
+            if (pluginsConfig == null)
+            {
+                return ArrayHelper.Empty<IOTelPlugin>();
+            }
+
+            var runtimeName = GetPluginsRuntimeName();
```

Comment:
This doesn't seem like the best way to get the name of the runtime, especially if running in the GAC will be a scenario. What about using the `FrameworkDescription.Instance` to get the name?